src/lazy/expanded/struct.rs

```diff
@@ -284,7 +285,7 @@ impl<'top, D: Decoder> LazyExpandedStruct<'top, D> {
             }
             // For any other kind of struct, do a linear scan over its fields until we encounter
             // one with the requested name.
-            _ => {
+            ValueLiteral(..) | MakeField(..) | MakeStruct(..) => {
```

Comment:
🪧 This change addresses [feedback from the previous PR ](https://github.com/amazon-ion/ion-rust/pull/878#discussion_r1880885723) by making the remaining kinds of struct explicit.